networkx/algorithms/dag.py

```diff
@@ -32,69 +29,35 @@
     "dag_to_branching",
 ]
 
-chaini = chain.from_iterable
-
-
-def descendants(G, source):
-    """Returns all nodes reachable from `source` in `G`.
-
-    Parameters
-    ----------
-    G : NetworkX Graph
-    source : node in `G`
-
-    Returns
-    -------
-    set()
-        The descendants of `source` in `G`
-
-    Raises
-    ------
-    NetworkXError
-        If node `source` is not in `G`.
-
-    Examples
-    --------
-    >>> DG = nx.path_graph(5, create_using=nx.DiGraph)
-    >>> sorted(list(nx.descendants(DG, 2)))
-    [3, 4]
+_deprecated_names = [
+    "descendants",
+    "ancestors",
+    "transitive_closure",
+]
 
-    See also
-    --------
-    ancestors
-    """
-    return {child for parent, child in nx.bfs_edges(G, source)}
 
+def __getattr__(name):
+    f = getattr(nx.algorithms.traversal.reachability, name)
```

Comment:
Ah - good idea, this is much lighter-weight than the idea I had originally proposed and I *think* this is okay since the full networkx namespace is imported in the module.

Reply:
Oh thank you :heart: